### PR TITLE
Remove Node v4 incompatible syntax

### PR DESCRIPTION
**Pull request by jhermsmeier**
- Fix Node v4 tests on Travis CI

### CODE REVIEW

**Comment by jviotti** [comment]
Any particular reason to support Node v4? ES6 introduces various features that can help us write better and more readable code (e.g: `const`, far arrow functions, etc), and I think we should use it if necessary.

Unless there is a strong reason to be 100% ES5 complaint, I suggests to go full ES6. If such a case appears I rather keep taking advantage of ES6 and transpile to ES5 as a build step (using Babel) or something.

What do you think?

**Comment by jhermsmeier** [comment]
Well, since Node v4 is LTS, and there are still some people on v0.12 (!), I think it's best to make sure it runs on as many versions & platforms as possible, especially since this is a public module with a fairly generic use case.

IMHO, adding Babel would be too much unnecessary overhead, compared to the benefits ES6 brings around (which I personally think aren't really any, but that's another story).

So, in conclusion, I'd either go full ES6 and drop Node v4 support, or stick with ES5 and just work everywhere. I'm happy with either way, if it makes work easier for us.

**Comment by lurch** [comment]
Given that this is in the `resin-io-modules` namespace, perhaps it makes sense to stick with the rest of the resin.io ecosystem and only support ES6?
(whereas if it was in the `jhermsmeier` namespace then perhaps it'd make more sense to still also support ES5?)

All IMVHO, IYSWIM ;-)

**Comment by jviotti** [comment]
> perhaps it makes sense to stick with the rest of the resin.io ecosystem and only support ES6?

The current resin.io ecosystem is a bit of a mess, with ES5, ES6, CoffeeScript, and TypeScript all over the place. I'd like to keep all Etcher specific stuff in ES6, so we share the exact same coding guidelines from the main Etcher repo, however I think this is a lower level module that is not tied to Etcher in any specific way, so let's keep it as ES5 for compatibility purposes also given you've made the work already :)

**Review by jviotti — APPROVED** [review]

**Comment by jviotti** [comment]
@jhermsmeier BTW, you're welcome to adopt/try TypeScript for some of these modules if you like.

**Comment by jhermsmeier** [comment]
Alright – yeah, I was thinking about maybe trying TypeScript some time – so far I've never used it, but it might come in handy with lower level things.

**Comment by jhermsmeier** [comment]
I also just realised that Node v8 will probably drop soon-ish, which means v6 will be the new LTS, and we could just drop v4 support and move to ES6 when that happens, if that makes sense?

**Comment by jviotti** [comment]
Sounds great to me!
